### PR TITLE
Algorithm to export ascii masks

### PR DESCRIPTION
**Pull request by abuts**
This is widely used in ISIS direct inelastic group algorithm which exports workspace masks, and saves them into ASCII file as special format file (.msk). The algorithm was used by group for ages and I just pull it out of qtiGenie and wrote Python algorithm wrapper, unit tests and documentation. As it is old working algorithm, I have not actually changed its core code.

**To test:**
Its small and simple algorithm, so look if the description is clear, all code conventions are satisfied and you can not break it by providing wrong data.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
When I run it from Mantidplot on the workspace created by the usage example. It fails to create a file in my default save directory, but does produce any error messages or warnings. I use Windows 7.

File: Framework/PythonInterface/plugins/algorithms/ExportSpectraMask.py [hunk 1]
```diff
@@ -170,12 +170,11 @@ def category(self):
...
         """ Return name
         """
-        return "ExportASCIIMask"
+        return "ExportSpectraMask"
 
     def summary(self):
-        return "Exports workspace masks as legacy ASCII file with .msk extension, "\
-            "containing list of masked spectra numbers, "\
-            "in the ISIS ASCII format described in LoadMask algorithm"
+        return "Returns list of spectra mnumbers which are masksed"\
```
Comment (KarlPalmen) [inline Framework/PythonInterface/plugins/algorithms/ExportSpectraMask.py]: "numbers" and "masked" have typos.

File: Framework/PythonInterface/plugins/algorithms/ExportSpectraMask.py [hunk 2]
```diff
@@ -184,10 +183,10 @@ def PyInit(self):
         self.declareProperty("Filename","","Name of the file to save mask to."\
                             " If empty, the name of the input workspace is used")
         self.declareProperty("ExportMaskOnly",False,"If true, algorithm will not save mask in a file"\
-                             "and returns only list with numbers of masted spectra.",\
+                             "and only returns the list containing numbers of masted spectra.",\
                               Direction.Input)
```
Comment (KarlPalmen) [inline Framework/PythonInterface/plugins/algorithms/ExportSpectraMask.py]: "masted" -> "masked".

**Comment by abuts** [comment]
All fixed -- it was using wrong target directory if just filename is provided.

**Comment by KarlPalmen** [comment]
Works OK and looks good. :shipit:

**Comment by martyngigg** [comment]
This seems like a useful little routine for ISIS. Would it not be better as a different format option on `SaveMask`? It seems inconsistent that you use `SaveMask` for XML and then `ExportSpectraMasks` for ASCII.

**Comment by abuts** [comment]
I am hoping to get rid of Save ASCII masks and change to xml eventually. Unfortunately, it brings up the range of issues related to the fact that the definition of spectra-detector map is not a part of an instrument definition but is always 1:1 and  **bound to some implicit order**, which is wrong for all ISIS instruments. Number of steps is necessary to overcome this and I am making some now. 

This is why I consider this algorithm as a temporary helper rather then permanent solution and as SaveMask is C++ and this one is existing Python, converting to C++ (or calling Python from C) at the moment seems too much hustle. Will see how people use it but its possible that it will just eventually die.

File: docs/source/algorithms/ExportSpectraMask-v1.rst [hunk 3]
```diff
@@ -14,6 +14,12 @@ if requested, saves these spectra as legacy *.msk* file in ISIS ASCII format.
 
 The format is described on :ref:`LoadMask <algm-LoadMask>` algorithm pages.
 
+**Note** It is recommended not use exported *.msk* files for masking workspaces during reduction as
+the result of applying such mask will be different depending on the stage of reduction of the source and target files.
+You need to make additional efforts (see :ref:`LoadMask <algm-LoadMask>` algorithm description) to avoid this effect.
+It is better to use :ref:`SaveMask <algm-SaveMask>` algorithm instead. This algorithm produces *.hml* files, 
+containing masked detectors lists, which will produce identical results regardless of the workspace grouping.
```
Comment (KarlPalmen) [inline docs/source/algorithms/ExportSpectraMask-v1.rst]: Did you mean '.hml' files?  I thought they were '.xml' files.

Comment (abuts) [inline docs/source/algorithms/ExportSpectraMask-v1.rst]: yep, right note -- fixed...

**Comment by martyngigg** [comment]
Now that the note has been added I think this can go in.

**Pull request merged**